src/elm/Auth.elm

```diff
@@ -230,23 +233,18 @@ type alias PrivateKey =
 isAuth : Model -> Bool
 isAuth model =
     case model.status of
-        LoggedInWithPin _ ->
-            True
-
-        LoggedInWithPrivateKey _ ->
+        WithPrivateKey _ ->
             True
 
+        -- TODO - Should `WithoutPrivateKey` be considered authed here? (Check usage in other modules)
```

Comment:
Yeah, semantically it makes sense to consider the user authenticated, just need to make sure the modules that use this function use it accordingly (before this change, only the states that had a `PrivateKey` were considered authenticated). 

Maybe renaming it would be more appropriate, since other modules basically use this to check if they need to call `LoggedIn.RequiredAuthentication`, which asks the user for their PIN.

Reply:
Also, thinking about it, I don't really like the structure we have right now. We're kind of duplicating the authentication state with `Guest`/`LoggedIn` and the `Auth` module. As far as I can tell, a `Guest` can only ever have an `Auth.Model` that is `Unauthenticated`, and (if I'm not mistaken) that's only used in the `Login` page (which at least in this PR is just a blank page that wraps around `Auth`). Similarly, a `LoggedIn` user is always either `WithPrivateKey` or `WithoutPrivateKey`, and the view for those is completely different from `Unauthenticated`. The only thing that's shared between the authenticated and unauthenticated states is input validation, input messages and a few helper functions.

Not sure what the best solution is, but one approach would be to get rid of the `Auth` module (at least the data modelling part, we could still reuse the validation and the views), and change the `Auth.Model` for a `Maybe PrivateKey` field in `LoggedIn.Model`. IMO that would be an equivalent model to the one we have now, but a lot easier to understand, especially to new people that aren't familiar with the way we're doing authentication (it actually took me a while to realize how we were getting the `PrivateKey` :sweat_smile:).

Reply:
hehehe I love it bro!

Ok, lets do it!

## Old Context
At some point the app was very flexible and allowed for a very different usage than we have right now. It allow any user to use the app freely without having a private key, only asking for it when it needed signing a transaction (we were backend free by then). It also supported some sort of login in, it was not as we have it today, as it only needed to have _something_ in the local storage to be considered logged, no private key required. With that "middle" state of logged without a private key you could do anything, except sign a transaction. 

You could also have a variance of that, allowing the key to be stored by setting a PIN, or adding it only to the next transaction (it would be discarded afterwards). It even supported Guest pages, public pages that could be seem by anyone. Community screen was one of those screens.

## New context
Ok, history class over. The app change drastically to have a backend, and now we **require** backend authentication and require PIN and private key to be able to use the app. 

Effectively we have some pages that can be seem as a guest (maybe we can call them public and restricted instead of guest and logged)

- Payments page
- Invitation pages
- Register pages
- Sign in page

all the rest are restricted. 

## New requirements
I'm ok on us removing Auth, but we need to actually have more info than just the private key. Private key is the base but we also need a way to know what's up with the authentication token. Check the validity of the token and only progress if it is ok. In every full page load, and even periodically we will need to check this. Imagine the app hasn't checked the auth token in a while, or got a 403 from the backend saying the token isn't valid anymore while doing some other requests. The app will have to know this, and using the private key renew the auth token. All of this need to be transparent to the user.

I agree with you @NeoVier that this part of the app is obscure, over complicated and a big technical debt that we need to sort out. The tradeoffs are incredible and necessary at this point!

Changing to this strategy also makes easier for us to use Natus' approach to authentication, based off email, creating users and relating them to one private key and one account, that can be changed and recovered.